src/cargo/sources/registry/remote.rs

```diff
@@ -113,8 +102,9 @@ impl<'cfg> RemoteRegistry<'cfg> {
             }
         }
         let repo = self.repo()?;
-        let commit = repo.find_commit(self.head()?)?;
-        let tree = commit.tree()?;
+        let oid = self.index_git_ref.resolve(repo)?;
+        let obj = repo.find_object(oid, None)?;
+        let tree = obj.peel_to_tree()?;
```

Comment:
We used to cache this lookup.  I'm assuming this optimization isn't needed anymore?

Reply:
We cache the call to `tree()` in `self.tree`, so it will only be called once.